rotkehlchen/data_handler.py

```diff
@@ -162,12 +162,7 @@ def remove_ignored_assets(self, assets: List[Asset]) -> Tuple[Optional[List[Asse
     def should_save_balances(self) -> bool:
         """ Returns whether or not we can save data to the database depending on
         the balance data saving frequency setting"""
-        last_save = self.db.get_last_balance_save_time()
-        settings = self.db.get_settings()
-        # Setting is saved in hours, convert to seconds here
-        period = settings.balance_save_frequency * 60 * 60
-        now = Timestamp(int(time.time()))
-        return now - last_save > period
+        return can_save_balances(self.db)
```

Comment:
This function has become a wrapper of `can_save_balances`, it would be better to call `can_save_balances `directly`. Also personally the name `should_save_balances` looks better than `can_save_balances` for what the function does since there is nothing stopping you from doing it but you shouldn't do it periodically